### PR TITLE
Strategy filter pattern

### LINKED ISSUE #79: Strategy pattern for filters
Implement strategies for optimization of filters.

### PR DESCRIPTION
**Pull request by ZivanicBojan**
Closes #79 
Implemented filter and search strategy pattern. Each filter model has its own strategy that is used when search and filter is called.

### CODE REVIEW

File: graph_visualizer_api/src/graph_visualizer_api/model/filter.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from __future__ import annotations
-
+from graph_visualizer_platform.filter_strategies import FilterStrategy
```
Comment (vradovic) [inline graph_visualizer_api/src/graph_visualizer_api/model/filter.py:2]: API should not depend on platform.

File: graph_visualizer_platform/src/graph_visualizer_platform/filter_strategies.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class FilterStrategy(ABC):
```
Comment (vradovic) [inline graph_visualizer_platform/src/graph_visualizer_platform/filter_strategies.py:6]: I would move this to API.

**Review by vradovic — CHANGES_REQUESTED** [review]
Looks good but API should not import from platform.

**Review by vradovic — APPROVED** [review]
Looks good!

**Review by trami25 — APPROVED** [review]
Thank you very much for coworking with me! Great job!

**Pull request merged**